consensus/consensus.go

```diff
@@ -445,3 +401,118 @@ func NewGenesisStakeInfoFinder() (*GenesisStakeInfoFinder, error) {
 	}
 	return f, nil
 }
+
+// Start is the entry point and main loop for consensus
+func (consensus *Consensus) Start(stopChan chan struct{}, stoppedChan chan struct{}) {
+	defer close(stoppedChan)
+	tick := time.NewTicker(blockDuration)
+	consensus.idleTimeout.Start()
+	for {
+		select {
+		default:
+			msg := consensus.recvWithTimeout(receiveTimeout)
+			consensus.handleMessageUpdate(msg)
+			if consensus.idleTimeout.CheckExpire() {
+				consensus.startViewChange(consensus.consensusID + 1)
+			}
+			if consensus.commitTimeout.CheckExpire() {
+				consensus.startViewChange(consensus.consensusID + 1)
+			}
+			if consensus.viewChangeTimeout.CheckExpire() {
+				if consensus.mode.Mode() == Normal {
+					continue
+				}
+				consensusID := consensus.mode.ConsensusID()
+				consensus.startViewChange(consensusID + 1)
+			}
+		case <-tick.C:
+			consensus.tryPublishBlock()
```

Comment:
shouldn't you start the idleTimeout after the block published? otherwise, you may lose some time for consensus.